root/utility/tableColumns.js

```diff
@@ -10,27 +10,82 @@
 import * as React from 'react';
 import type {ColumnOptions} from 'react-table';
 
+import SortableTableHeader from '../components/SortableTableHeader';
+import DescriptiveLink
+  from '../static/scripts/common/components/DescriptiveLink';
 import EntityLink from '../static/scripts/common/components/EntityLink';
 import yesNo from '../static/scripts/common/utility/yesNo';
 
-export function defineNameColumn(
+export function defineCheckboxColumn<T>(
+  name: string,
+): ColumnOptions<EntityRoleT<T>, number> {
+  return {
+    Cell: ({cell: {value}}) => (
+      <input
+        name={name}
+        type="checkbox"
+        value={value}
+      />
+    ),
+    Header: <input type="checkbox" />,
+    accessor: 'id',
+    className: 'checkbox-cell',
+    id: 'checkbox',
+  };
+}
+
+export function defineNameColumn<T: CoreEntityT | CollectionT>(
+  descriptive: boolean,
+  order: ?string,
+  sortable: ?boolean,
   title: string,
-): ColumnOptions<CoreEntityT | CollectionT, string> {
+): ColumnOptions<T, string> {
   return {
-    Cell: ({row: {original}}) => (
-      <EntityLink entity={original} />
+    Cell: ({row: {original}}) => (descriptive
+      // $FlowFixMe
+      ? <DescriptiveLink entity={original} />
+      : <EntityLink entity={original} />
     ),
-    Header: title,
+    Header: (sortable
+      ? (
+        <SortableTableHeader
+          label={title}
+          name="name"
+          order={order}
+        />
+      )
+      : title),
     accessor: 'name',
+    id: 'name',
   };
 }
 
-export const typeColumn: ColumnOptions<CollectionT, string> = {
-  Cell: ({cell: {value}}) => l(value),
-  Header: N_l('Type'),
-  accessor: 'typeName',
-  id: 'type',
-};
+export function defineTypeColumn(
+  order: ?string,
+  sortable: ?boolean,
+  typeContext: string,
+): ColumnOptions<CoreEntityT | CollectionT, string> {
+  return {
+    Cell: ({cell: {value}}) => (value
+      ? (typeContext
+        ? lp_attributes(value, typeContext)
+        : l(value)
+      )
+      : null),
+    Header: (sortable
+      ? (
+        <SortableTableHeader
+          label={l('Type')}
+          name="type"
+          order={order}
+        />
+      )
+      : l('Type')),
+    // $FlowFixMe
+    accessor: 'typeName',
```

Comment:
Issue is `CoreEntityT` should probably be replaced with something else here as it doesn’t have a `typeName`.